Dev Ticket Flow + Bot DevOps

cha0sg0d opened the pull request:
https://share.cleanshot.com/Ps3D6hz5

- [x] [Instructions](https://github.com/emergenceland/zupass/blob/test/bot.md) for running the TG <> PCDPass auth / join flow end to end.
- [x] Add @saurfang's `/setup` and scratch ticket cli https://github.com/saurfang/zupass/tree/forestfang--scratch-cli commands
- [x] Add `SERVER_EDDSA_PRIVATE_KEY` to `.env.local.example` for easier quick start
- [x] Automate adding the `channel_id` to the `telegram_bot_conversations` table (`/link`) command
- [x] Shoutout @lermchair for making local https cert give us beautiful local dev experience for TG Web App
- [x] Close Web App on success

- [ ] Figure out the best way to determine redirect logic for `/telegram/verify`. If user is coming from a website or mobile web, we want to redirect to https://t.me/bot_name. If user is coming from the TG Web App, we want to send some custom HTML and close the Web App.
- [ ] Figure out why `chat_join_request` isn't being triggered when a user goes to join the chat.

saurfang committed:
add /setup tg command

saurfang committed:
local dev events and tickets cli

robknight committed:
Generic issuance (#516)

Implements feeds, allowing third parties to make PCDs available to `passport-client`.

In order to resolve build issues, `consumer-client` has been converted to a regular React SPA, away from a NextJS app.

cha0sg0d committed:
feat: command to just dev passport server and client

cha0sg0d committed:
feat: .env setup

cha0sg0d committed:
feat: confirm local tickets work

cha0sg0d committed:
chore: yarn

cha0sg0d committed:
chore: yarn.lock

cha0sg0d committed:
chore: merge cleanup

cha0sg0d committed:
chore: docs

cha0sg0d committed:
chore: clean up

cha0sg0d committed:
feat: /link command to remove any postico manipulation

cha0sg0d committed:
feat: local https works lerm the goat

cha0sg0d committed:
chore: docs for local https

cha0sg0d committed:
feat: close web view on success + docs

saurfang commented on apps/passport-server/src/database/queries/pretixEventInfo.ts:
```diff
@@ -30,6 +30,7 @@ export async function insertPretixEventsInfo(
     `\
       insert into devconnect_pretix_events_info (event_name, pretix_events_config_id, checkin_list_id, is_deleted)
       values ($1, $2, $3, FALSE)
+      on conflict (pretix_events_config_id) do update set event_name = $1, checkin_list_id = $3
```
note: I didn't examine the data schema in details and came up with these on conflict clause based on common sense. In production, this will turn some queries that would otherwise fail to update/overwrite existing records. We should make sure that are safe. 

Alternatively, these functions can gain a "update in place" flag that kicks in on demand.

cha0sg0d committed:
chore: remove eslint-disable for IS_LOCAL_HTTPS

cha0sg0d committed:
chore: use port variable in server.ts

cha0sg0d committed:
chore: remove space in telegramService.ts comment

cha0sg0d committed:
chore: reduce some hardcoding via IS_LOCAL_SERVER variable in telegramService.ts

cha0sg0d committed:
chore: concurrently

cha0sg0d committed:
feat: fix IS_LOCAl_SERVER var and remove setup command

cha0sg0d committed:
chore: reorder event info check

cha0sg0d committed:
chore: richard pr edits

cha0sg0d committed:
feat: link to hot local https explanations

cha0sg0d committed:
chore: fix test event name

cha0sg0d committed:
feat: new dev quick start

cha0sg0d committed:
chore: help when yarn scratch is run incorrectly

cha0sg0d committed:
chore: remove on_conflict handling from eventInfo and itemInfo

saurfang committed:
fix flickering background color (#603)

When loading PCDPass app, we always get the `--bg-dark-primary` bg color
from the static css file first. However, many screens use `gray` bg
colors and it creates a odd experience when loading a path directly. For
example, opening the /prove link from Telegram, one would see a
flickering background color.

If we simply remove the background color in the stylesheet, the
background color ended up to be a flash of white on first paint because
AppContainer is not yet loaded. Indeed, this is a common problem
especially for websites with dark mode:
https://stackoverflow.com/questions/37126401/modify-bodys-classes-before-first-paint

This PR introduces an eagerly executed js blob that pattern matches on
location hash. This ensure the bg color is correctly set on first paint.
It also refactor the AppContainer bg to switch on a centralized list of
routes config. We still want to keep the AppContainer injection because
the hand written js might not always give the correct result as the
HashRouter.

Tested locally with both `yarn dev` and `yarn build` to ensure build can
access the config file.

See the comparison between this branch and production build



https://github.com/proofcarryingdata/zupass/assets/4317392/6d952d0b-27a7-4b64-a53f-2029f19c187a

relates to https://github.com/proofcarryingdata/zupass/issues/597

---------

Co-authored-by: Forest Fang <4317392+saurfang@users.noreply.github.com>

ichub committed:
fix conflict caused by #603

cha0sg0d committed:
feat: handle conflicts for telegram_bot_events

cha0sg0d committed:
test: fetchEventInfoByName

ichub committed:
Revert "fix conflict caused by #603"

This reverts commit 5891af277f1d6650c61d3828eba1d0e1de87f03b.

ichub committed:
Revert "fix flickering background color (#603)"

This reverts commit 8e516f2fa51744b8d025c861284e774b9154990f.

cha0sg0d committed:
feat: test new logic for telegram bot events. telegramService needs work now

cha0sg0d committed:
chore: yarn.lock after rebase

cha0sg0d committed:
chore: dummy

cha0sg0d committed:
chore: fix yarn.lock

cha0sg0d committed:
chore: tests

cha0sg0d committed:
Update bot.md

Co-authored-by: Richard Liu <richard.y.liu@berkeley.edu>

cha0sg0d committed:
Remove public from table IDs

Co-authored-by: Richard Liu <richard.y.liu@berkeley.edu>

cha0sg0d committed:
chore: isLocalServer util and require an event name for /link command

cha0sg0d committed:
chore: todo for tg bot Menu and docs

cha0sg0d committed:
chore: bot.md formatting

cha0sg0d committed:
chore: rebase

cha0sg0d committed:
chore: util

cha0sg0d committed:
chore: auto verify in staging (will be fixed in future PR)

cha0sg0d committed:
Apply .env.local.example suggestions

Co-authored-by: Richard Liu <richard.y.liu@berkeley.edu>

cha0sg0d committed:
chore: issue for fetchEventInfo and passport:dev

cha0sg0d committed:
chore: passport:dev command

cha0sg0d committed:
chore: remove PRETIX_ORG_URL from cli dev event

cha0sg0d committed:
fix: remove faulty migration
